pytorch_lightning/plugins/environments/slurm_environment.py

```diff
@@ -20,64 +20,154 @@
 
 
 class SLURMEnvironment(ClusterEnvironment):
+    """An environment for running on clusters managed by the LSF resource manager.
 
-    def __init__(self):
-        super().__init__()
+    It is expected that any execution using this ClusterEnvironment was executed
+    using the srun
 
-    def master_address(self):
-        # figure out the root node addr
-        slurm_nodelist = os.environ.get("SLURM_NODELIST")
-        if slurm_nodelist:
-            root_node = slurm_nodelist.split(" ")[0].split(",")[0]
-        else:
-            root_node = "127.0.0.1"
+    This plugin expects the following environment variables:
+
+    SLURM_JOB_ID
+      The Slurm assigned job ID
+
+    SLURM_NODELIST
+      The hosts used in the job. This string is expected to have the format "<rank_0_host> ...."
 
-        root_node = self._resolve_root_node_address(root_node)
-        os.environ["MASTER_ADDR"] = root_node
+    SLURM_LOCALID
+      The node local rank for the task.
+
+    SLURM_PROCID
+      The MPI rank or relative process ID
+
+    SLURM_STEP_NUM_TASKS
+      The world size for the job. This environment variable is set by srun
+    """
```

Comment:
nice docs! 
It would be awesome if you could break this out into a separate PR :)

Reply:
Yeah, I'm happy to separate this.